gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/orc/OrcUtils.java

```diff
@@ -482,4 +483,32 @@ public static WritableComparable createValueRecursively(TypeDescription schema,
   public static WritableComparable createValueRecursively(TypeDescription schema) {
     return createValueRecursively(schema, 1);
   }
+
+  /**
+   * Check recursively if owning schema is eligible to be up-converted to targetSchema:
+   * 1. TargetSchema is a subset of originalSchema.
+   * 2. TargetSchema is larger than originalSchema.
+   */
+  public static boolean eligibleForUpConvert(TypeDescription originalSchema, TypeDescription targetSchema) {
+    if (!targetSchema.getCategory().isPrimitive()) {
+      if (originalSchema.getFieldNames().size() < targetSchema.getFieldNames().size()) {
+        return true;
```

Comment:
Why is targetSchema having more fields than original schema a sufficient condition for up conversion?

Reply:
Shouldn't we be checking if each field in the target Schema is present in the record schema?